### PR TITLE
Do not produce KubernetesClient if we have OpenShiftClient available

### LINKED ISSUE #14943: Kubernetes clients all clash with one another
When importing the `quarkus-openshift-client` extension, we get an injectable `OpenShiftClient` which also extends `KubernetesClient` which is located in the transitive dependency `quarkus-kubernetes-client`, we get a CDI clash on any `KubernetesClient` injection points, because we now have two clients which implement it.

We could fix this by disabling the producer of `KubernetesClient` if the `OpenShiftClient` producer exists, but this won't scale for the other types of clients (there are 3 subtypes of `KubernetesClient` from 3 separate extensions that can all be imported).

So I think the best is to restrict the produced type of `OpenShiftClient` so that it does not clash, and users can inject either type of client separately.

### PR DESCRIPTION
**Pull request by FroMage**
Fixes #14943

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Lovely!

**Comment by gsmet** [comment]
GH Actions went berserk on this PR. We have a ton of infra failures. I will restart it later.

**Comment by FroMage** [comment]
OK thanks.

**Comment by gsmet** [comment]
Update - GitHub Actions is still experiencing degraded performance. We are still investigating and will provide an update when we have one.
Feb 9, 16:25 UTC

**Comment by FroMage** [comment]
Hey man, you don't have to justify github's failures, do what you gotta do ;)

**Comment by gsmet** [comment]
Restarted CI.

File: integration-tests/openshift-client/src/main/java/io/quarkus/it/openshift/client/Routes.java [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
-    private final OpenShiftClient openshiftClient;
+    // make sure we can inject both aspects of it
+    @Inject
+    KubernetesClient kubernetesClient;
 
-    public Routes(OpenShiftClient openshiftClient) {
-        this.openshiftClient = openshiftClient;
-    }
+    @Inject
+    OpenShiftClient openshiftClient;
```
Comment (gastaldi) [inline integration-tests/openshift-client/src/main/java/io/quarkus/it/openshift/client/Routes.java:20]: Can you add a test that `assertSame(kubernetesClient, openshiftClient)`?

Comment (FroMage) [reply integration-tests/openshift-client/src/main/java/io/quarkus/it/openshift/client/Routes.java:20]: Done.

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**